Mathlib/Algebra/Order/Floor.lean

```diff
@@ -1200,6 +1233,46 @@ theorem ceil_sub_self_eq (ha : fract a ≠ 0) : (⌈a⌉ : α) - a = 1 - fract a
   rw [(or_iff_right ha).mp (fract_eq_zero_or_add_one_sub_ceil a)]
   abel
 
+section LinearOrderedField
+variable {k : Type*} [LinearOrderedField k] [FloorRing k] {a b : k}
+
+lemma ceil_div_ceil_inv_sub_one (ha : 1 ≤ a) : ⌈⌈(a - 1)⁻¹⌉ / a⌉ = ⌈(a - 1)⁻¹⌉ := by
+  obtain rfl | ha := ha.eq_or_lt
+  · simp
+  have : 0 < a - 1 := by linarith
+  have : 0 < ⌈(a - 1)⁻¹⌉ := ceil_pos.2 <| by positivity
```

Comment:
positivity can't use ceil_pos?

Reply:
It can, but only after the end of the file